elastic/kibana
# [Fleet] Task to publish Agent metrics

## Summary

Closes https://github.com/elastic/ingest-dev/issues/2396

Added a new kibana task that publishes Agent metrics every minute to data streams installed by fleet_server package.

Opened the pr for review, there are a few things to finalize, but the core logic won't change much.

To test locally:
- Install fleet_server package 1.4.0 from [this](https://github.com/elastic/integrations/pull/8145) pr to get the mappings
- Start kibana locally, wait for a few minutes for the metrics task to run (every minute)
- Go to discover, `metrics-*` index pattern, filter on `data_stream.dataset: fleet_server.*`
- Expect data to be populated in `fleet_server.agent_status` and `fleet_server.agent_versions` datasets.

<img width="1787" alt="image" src="https://github.com/elastic/kibana/assets/90178898/615af9df-fe4b-4c17-8c8c-88646c403a18">



### Checklist
